tools/crm_resource_runtime.c

```diff
@@ -1730,7 +1730,8 @@ cli_resource_execute_from_params(pcmk__output_t *out, const char *rsc_name,
     params_copy = pcmk__str_table_dup(params);
 
     op = resources_action_create(rsc_name ? rsc_name : "test", rsc_class, rsc_prov,
-                                 rsc_type, action, 0, timeout_ms, params_copy, 0);
+                                 rsc_type, pcmk__starts_with(action, "force-") ?
+                                 "validate-all" : action, 0, timeout_ms, params_copy, 0);
```

Comment:
We would actually need the checks at the start of cli_resource_execute() (mapping an option to an action). I think the easiest thing will be to just move those from cli_resource_execute() to cli_resource_execute_from_params(), so the latter takes an option name rather than an action name, and maps it to an action internally.

Reply:
Done.